libraries/GCS_MAVLink/GCS_FTP.cpp

```diff
@@ -609,7 +609,12 @@ int GCS_MAVLINK::gen_dir_entry(char *dest, size_t space, const char *path, const
 #endif
         const size_t full_path_len = strlen(path) + strnlen(entry->d_name, max_name_len);
         char full_path[full_path_len + 2];
-        hal.util->snprintf(full_path, sizeof(full_path), "%s/%s", path, entry->d_name);
+        if (strcmp(path, "/") == 0) {
```

Comment:
any change like this would have to be in AP_Filesystem_posix.cpp so it doesn't affect SYS, ROMFS, MISSION etc